File: JANL/Designers/HResizeOnly.vb
Imports System.Windows.Forms.Design

Public Class HResizeOnly
	Inherits ControlDesigner

	Public Sub New()
		AutoResizeHandles = True
	End Sub

	Public Overrides ReadOnly Property SelectionRules() As SelectionRules
		Get
			Return CType(SelectionRules.LeftSizeable + SelectionRules.RightSizeable + SelectionRules.Moveable, SelectionRules)
		End Get
	End Property

End Class

File: JANL.WinForms/Designers/HResizeOnly.cs
using System.Windows.Forms.Design;

namespace JANL.Designers
{
    /// <summary>
    /// Разрешает изменение только ширины
    /// </summary>
    public class HResizeOnly : ControlDesigner
    {
        /// <summary>
        ///
        /// </summary>
        public HResizeOnly()
        {
            AutoResizeHandles = true;
        }

        /// <summary>
        /// Правила изменения размера
        /// </summary>
        public override SelectionRules SelectionRules => SelectionRules.LeftSizeable | SelectionRules.RightSizeable | SelectionRules.Moveable;
    }
}